proposals/gc/MVP.md

```diff
@@ -285,16 +277,20 @@ Perhaps also the following short-hands:
 
 #### Runtime Types
 
-* `rtt.get <typeuse>` returns the RTT of the specified type
-  - `rtt.get t : [] -> [(rtt t)]`
+* `rtt.none` returns the RTT that cannot be used to cast to anything
+  - `rtt.none : [] -> [(rtt anyref)]`
   - multiple invocations of this instruction yield the same observable RTTs
+
+* `rtt.new <typeuse>` returns the RTT of the specified type
+  - `rtt.new t : [] -> [(rtt t)]`
+  - each invocation of this instruction yields a distinct RTT
+  - equivalent to `(rtt.sub t (rtt.none))`
   - this is a *constant instruction*
-  - equivalent to `(rtt.sub t (rtt.get anyref))`, except when `t` itself is `anyref`
 
 * `rtt.sub <typeuse>` returns the RTT of the specified type as a sub-RTT of a given parent RTT operand
   - `rtt.sub t : [(rtt t')] -> [(rtt t)]`
     - iff `t <: t'`
-  - multiple invocations of this instruction with the same operand yield the same observable RTTs
+  - each invocation of this instruction yields a distinct RTT
   - this is a *constant instruction*
```

Comment:
Also, this assertion was already present but wasn't true. For an applicative semantics, one must canonicalize the structure of the type (which is not constant time), hash it (again not constant time), and then use a host-global concurrent hash map to get the canonical `rtt` for the type (requiring another non-constant-time equality comparison). When the type is known it compile time, which is *currently* the case for this proposal, then all of that can be done at compile time and the instruction replaced with the precomputed `rtt`, which is where the assertion that this instruction is constant time comes from. However, the proposal also recommends adding parametric polymorphism later on, which will introduce type parameters that the type in `rtt.get` can refer to, in which case all of this will have to be done at run time and the instruction will no longer be constant time. With the generative semantics, this is constant time even in the presence of type variables.

Reply:
"Constant instruction" is [spec terminology](https://webassembly.github.io/spec/core/valid/instructions.html#constant-expressions) that has nothing to do with constant time.

Reply:
Ah, thanks for the correction and the link!